src/ucs/config/parser.c

```diff
@@ -1862,19 +1872,26 @@ static void ucs_config_parser_warn_unused_env_vars(const char *prefix)
 
         iter = kh_get(ucs_config_env_vars, &ucs_config_parser_env_vars, var_name);
         if (iter == kh_end(&ucs_config_parser_env_vars)) {
-            ret = snprintf(p, endp - p, " %s,", var_name);
-            if (ret > endp - p) {
-                truncated = 1;
-                *p = '\0';
-            } else {
-                p += strlen(p);
-                ++num_unused_vars;
+            if (ucs_global_opts.warn_unused_env_vars) {
+                ret = snprintf(p, endp - p, " %s,", var_name);
+                if (ret > endp - p) {
+                    truncated = 1;
+                    *p = '\0';
+                } else {
+                    p += strlen(p);
+                    ++num_unused_vars;
+                }
             }
+        } else {
+            ucs_string_buffer_appendf(&strb, "%s, ", *envp);
```

Comment:
i'd remove the comma, to make more space for env vars

Reply:
maybe leave comma, but remove space?

Reply:
I agree with @yosefe: in case if output string is looks like ```VAR1=val1 VAR2=val2``` then it may be copied into any script to set runtime parameters as-is, without transformations

Reply:
ok, i see the point